internal/pkg/runtime/engine/singularity/prepare_linux.go

```diff
@@ -369,60 +369,78 @@ func (e *EngineOperations) prepareRootCaps() error {
 	return nil
 }
 
-func (e *EngineOperations) prepareFd(starterConfig *starter.Config) error {
+func keepAutofsMount(source string, autoFsPoints []string) (int, error) {
+	resolved, err := filepath.EvalSymlinks(source)
+	if err != nil {
+		return -1, err
+	}
+
+	for _, p := range autoFsPoints {
+		if strings.HasPrefix(resolved, p) {
+			sylog.Debugf("Open file descriptor for %s", resolved)
+			f, err := os.Open(resolved)
+			if err != nil {
+				return -1, err
+			}
+			return int(f.Fd()), nil
+		}
+	}
+
+	return -1, fmt.Errorf("no mount point")
+}
+
+func (e *EngineOperations) prepareAutofs(starterConfig *starter.Config) error {
+	const mountInfoPath = "/proc/self/mountinfo"
+
+	entries, err := proc.GetMountInfoEntry(mountInfoPath)
+	if err != nil {
+		return fmt.Errorf("while parsing %s: %s", mountInfoPath, err)
+	}
+	autoFsPoints := make([]string, 0)
+	for _, e := range entries {
+		if e.FSType == "autofs" {
+			sylog.Debugf("Found %q as autofs mount point", e.Point)
+			autoFsPoints = append(autoFsPoints, e.Point)
+		}
+	}
+	if len(autoFsPoints) == 0 {
+		sylog.Debugf("No autofs mount point found")
+		return nil
+	}
+
 	fds := make([]int, 0)
 
 	if e.EngineConfig.File.UserBindControl {
 		for _, b := range e.EngineConfig.GetBindPath() {
 			splitted := strings.Split(b, ":")
 
-			src, err := filepath.Abs(splitted[0])
-			if err != nil {
-				continue
-			}
-
-			if !fs.IsDir(src) {
-				continue
-			}
-
-			sylog.Debugf("Open file descriptor for %s", src)
-			f, err := os.Open(src)
-			if err != nil {
-				continue
+			if fd, err := keepAutofsMount(splitted[0], autoFsPoints); err == nil {
```

Comment:
Should this fail (return an error) if one occurs?

Reply:
All errors are deliberately ignored, we are in a best-effort approach regarding autofs bug, Singularity will do its best to fix that transparently for users but must not abort if it fail, if we can't open file for some reason (eg: permissions rights), I could log those errors to help further debugging